### DISCUSSION TITLE
Automate Creation of Calculation Groups for Time Intelligence

### DISCUSSION THREAD

**Discussion by patrickbcox** [TabularEditor/TabularEditor3#1275] [🙋♀️TE3 Q&A] [answered]
I have a standard Time Intelligence calculation group that I typically use for my models.  Is there a way to automate the creation of this Calculation Group and items in new models?  Thanks.

**Comment by mlonsk** [comment, marked as answer]
Hi @patrickbcox 
The simplest way is to right click on your calculation group and use the [DAX Script feature](https://docs.tabulareditor.com/te3/features/dax-scripts.html) to save a DAX Script that you then open in the new model and apply.

If you want to automate it more is to create a C# script that creates the Time intelligence calculation group, make it into a [macro](https://docs.tabulareditor.com/te3/tutorials/creating-macros.html) and reuse it your next model. Bernat Agullo has an excellent example [here](https://www.esbrina-ba.com/time-intelligence-the-smart-way)

If you wish to go fully automatic you can also look into using the C# script together with the TE2 CLI and run it over existing models.

**Comment by patrickbcox** [comment]
Thanks.  I will try that.
